source/maze/MazeIO.cpp

```diff
@@ -29,12 +29,16 @@ MazeIO::MazeIO(Maze *mazePtr,  IOInterface *fileIO, IOInterface *printIO) :
  ******************************************************************************/
 Position MazeIO::getMousePosition(void)
 {
-    return mousePosition;
+    return mousePosition.getCurrentPos();
 }
 
 void MazeIO::setMousePosition(Position pos)
 {
-    mousePosition = pos;
+    mousePosition = PositionController(pos, eDirError);
+}
+void MazeIO::setMousePosition(PositionController posCon)
```

Comment:
Please keep in mind the coding convention. One empty line between method definitions.